src/par_iter/enumerate.rs

```diff
@@ -89,6 +90,17 @@ pub struct EnumerateProducer<P> {
 impl<P> Producer for EnumerateProducer<P>
     where P: Producer
 {
+    type Item = (usize, P::Item);
+    type IntoIter = iter::Zip<Range<usize>, P::IntoIter>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        // Enumerate only works for ExactParallelIterators. Since those
+        // have a max length of usize::MAX, their max index is
+        // usize::MAX - 1, so the range 0..usize::MAX includes all
+        // possible indices
+        (self.offset..usize::MAX).zip(self.base.into_iter())
```

Comment:
I think `self.offset..` suffices, in that case, unless you wanted to make the `usize::MAX` clear for stylistic reasons.